pynhm/base/storageUnit.py

```diff
@@ -31,30 +29,11 @@ def __init__(
         self._output_netcdf = False
         self._netcdf = None
         self._separate_netcdf = True
-        self._itime_step = -1
-
-        # Go through list of parameters for this process and assign them
-        # to self with a value of None
-        for param in self.get_required_parameters():
-            setattr(self, param, None)
-
-        # Go through the parameters for this process and see if self
-        # has a variable with that name.  If so, then assign the parameter
-        # value to the self variable.
-        for key in params.parameters:
-            if hasattr(self, key):
-                setattr(self, key, params.parameters[key])
-
-        # if any of the required parameters are still none,
-        # then we should terminate with an error
-        for key in self.get_required_parameters():
-            value = getattr(self, key)
-            if value is None:
-                print(
-                    f"{storage_type} storage unit requires {key} but it was not found in parameters."
-                )
+        self._itime_step = -1  # JLM CHECK
 
-        return
+        self.initialize_self_variables()
```

Comment:
i streamlined with this: it sets the params, inputs, and variables on self in the base/StorageUnit class